test/jdk/javax/accessibility/4670319/AccessibleJTrePCESourceTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2008, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @key headful
+ * @bug 4670319
+ * @summary AccessibleJTree should fire a PropertyChangeEvent using a AccessibleJTreeNode as source.
+ * @run main AccessibleJTrePCESourceTest
+ */
+
+import java.awt.Robot;
+import java.beans.PropertyChangeEvent;
+import java.util.ArrayList;
+
+import javax.swing.JFrame;
+import javax.swing.JTree;
+import javax.swing.SwingUtilities;
+
+public class AccessibleJTrePCESourceTest {
+    private static JTree jTree;
+    private static JFrame jFrame;
+
+    private static ArrayList<PropertyChangeEvent> eventsList = new ArrayList<PropertyChangeEvent>();
+
+    private static void doTest() throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> createGUI());
+            Robot robot = new Robot();
+            robot.setAutoDelay(500);
```

Comment:
Are you sure that the autodelay is needed here? The test does not use any method other than `waitForIdle`